### ISSUE TITLE
tracked property (dependent on "args") in child component is out of sync

### ISSUE THREAD

**Issue by bantic** [glimmerjs/glimmer-component#66]
While attempting to follow along with the glimmerjs guides, @rondale-sc and I had some difficulty using `@tracked('args')` to create a derived computed property in a child component.

The guides suggest using the following to create a child component that has a computed property `multiplied` that updates when its `args.number` changes:

```javascript
import Component from '@glimmer/component';

export default class MultiplyByTwo extends Component {
  @tracked('args')
  get multiplied() {
    return this.args.number * 2;
  }
}
```
```hbs
<!-- my-parent/template.hbs -->
<multiply-by-two @number={{myNumber} />}
```
```hbs
<!-- multiply-by-two/template.hbs -->
{{multiplied}}
```

This seems to work as expected in the glimmer-playground (which appears to use `@glimmer/component` version 0.5.0). This is a link to a working version (with code lightly adapted from the code from the guides): http://bit.ly/2syFV7c

However, in a new glimmer app (created using ember cli with the `@glimmer/blueprint` blueprint), using `@glimmer/application` v0.7.2 and `@glimmer/component` v0.6.0, the same code doesn't work correctly. Clicking the buttons updates the property shown in the template, but not the derived `multiplied` property.

Here is a link to a commit from the app that exhibits this issue: 
https://github.com/bantic/bodega-glimmer-test/commit/f296b4ffd5ffbeac0ba9ba0fbb6461ce869ef244

Expected behavior: Clicking "number: 2" immediately changes the template to display "multiply: 4 (@number: 2)".
Actual behavior: On the first click, the `@number` text updates (to `"@number: 2"`), but the "multiply" text does not (i.e. `"multiply: 2 (@number: 2)"). Clicking the same button a second time causes the template to display the expected output.

![bug](https://user-images.githubusercontent.com/2023/27974791-7b1e2c84-632d-11e7-80a1-beb763650c92.gif)

**Comment by deepan83** [comment]
Hitting the same problem

**Comment by rwjblue** [comment]
FYI - https://github.com/glimmerjs/glimmer-component/pull/67 is attempting to address this issue.  I _believe_ that it is good to go, but we need another round of review...

**Comment by chrmod** [comment]
have exactly same issue here. is there a beta channel to test the fix? @rwjblue @tomdale

**Comment by tomdale** [comment]
@chrmod @deepan83 Ack, sorry, I didn't realize we hadn't cut a new release since @robbiepitts's fix for this. @chrmod You should be able to install the dependency from GitHub for the moment to verify whether it's fixing your issue.

I will cut a new release either today or tomorrow morning. We are trying to move `@glimmer/component` and `@glimmer/application` into a monorepo (https://github.com/glimmerjs/glimmer.js/pull/3) and this seems like a good test of the new system. I'll cut a new release once @rwjblue approves those changes, or just bite the bullet and do it with the existing setup if there are any problems with the PR that can't be immediately resolved.

**Comment by chrmod** [comment]
@tomdale thanks for the information. 

I've tried building and linking `@glimmer/component` master, but `each` helper does not seem to work - nothing gets rendered. It may be due to to some incompatibility with my version of `@glimmer/application` or other modules. 

I'm using glimmer without ember-cli (but with own, custom build system), so there may be other incompatible dependencies. To be honest it is very hard to figure out what exactly is need to build and run glimmer application, main reason is just the number of glimmer related repositories. So monorepo sounds like great idea for me.

**Comment by tomdale** [comment]
@chrmod I would bet linking is not working because you end up with two copies of `@glimmer/reference`. Tracked properties won't update correctly if you have duplicate `@glimmer/reference` packages because it uses a revision counter that is supposed to be global across the entire program. If you're not sharing that counter, Glimmer can't "see" the changes. If you `npm install` it into your app from GitHub, it should work, or you can make sure that both packages are using the same `@glimmer/reference` package.

**Comment by chrmod** [comment]
@tomdale this is all good to know, thank you again.

But how to npm install `@glimmer/component` from github? 
Specifying dependency in this format 
```
"@glimmer/component": "glimmerjs/glimmer-component",
```
fetches the project, but it is not build, so cannot be imported. 
I assume that running build in this form will result with exact same output as linking.

**Comment by chrmod** [comment]
@tomdale ok, I've figured it out. Had to `npm pack` local version of `glimmer-component` and install it with `npm install <path-to-glimmer-component.tgz>`.

Can confirm that https://github.com/glimmerjs/glimmer-component/pull/67 fixes the problem.  

Waiting for the release :smile:

**Comment by GavinJoyce** [comment]
I'm keenly awaiting a new release so that I can complete [my video tutorial on Glimmer.js](https://twitter.com/gavinjoyce/status/908972572662013952). I'd rather not have to mention anything about avoiding `0.7.2`.

@tomdale if the release is blocked for some reason, please LMK if there is anything you think I or others can do to help unblock it.

**Comment by tschoartschi** [comment]
It seems that I'm hitting the same problem. I created a repo, you can find it here: https://github.com/tschoartschi/glimmer-troubles I also created a screencast which is available here: https://github.com/tschoartschi/glimmer-troubles/blob/master/screencast.mov

**Comment by forbesmyester** [comment]
This:

```
export default class DecileGraph extends Component {
  @tracked
  public args;

  @tracked('args')
  get count(): number {
  ...
```

appeared to work but caused other problems... Is there any kind of work-around for this?

**Comment by tomdale** [comment]
@tschoartschi Can you verify the issue is resolved on 0.8.x of Glimmer.js?

**Comment by GavinJoyce** [comment]
I can confirm that the issue is fixed in 0.8:

 * [Broken in 0.7](http://glimmer-playground-beta.surge.sh/?app=JTVCJTdCJTIybmFtZSUyMiUzQSUyMm15LWdsaW1tZXItYXBwJTIyJTJDJTIydGVtcGxhdGUlMjIlM0ElN0IlMjJmaWxlTmFtZSUyMiUzQSUyMnNyYyUyRnVpJTJGY29tcG9uZW50cyUyRm15LWdsaW1tZXItYXBwJTJGdGVtcGxhdGUuaGJzJTIyJTJDJTIyc291cmNlVGV4dCUyMiUzQSUyMiUzQ2RpdiUzRSU1Q24lMjAlMjAlM0NoMSUzRUJ1ZyUyMGluJTIwJTQwZ2xpbW1lciUyRmFwcGxpY2F0aW9uJTIwMC43LjAlM0MlMkZoMSUzRSU1Q24lMjAlMjAlN0IlN0IlMjNlYWNoJTIwdmFsdWVzJTIwa2V5JTNEJTVDJTIyJTQwaW5kZXglNUMlMjIlMjBhcyUyMCU3Q3ZhbHVlJTIwaW5kZXglN0MlN0QlN0QlNUNuJTIwJTIwJTIwJTIwJTNDaW5uZXItY29tcG9uZW50JTVDbiUyMCUyMCUyMCUyMCUyMCUyMCU0MHZhbHVlJTNEJTdCJTdCdmFsdWUlN0QlN0QlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTQwb25DbGljayUzRCU3QiU3QmFjdGlvbiUyMGluY3JlbWVudEluZGV4JTIwaW5kZXglN0QlN0QlNUNuJTIwJTIwJTIwJTIwJTJGJTNFJTVDbiUyMCUyMCU3QiU3QiUyRmVhY2glN0QlN0QlNUNuJTNDJTJGZGl2JTNFJTIyJTdEJTJDJTIyY29tcG9uZW50JTIyJTNBJTdCJTIyZmlsZU5hbWUlMjIlM0ElMjJzcmMlMkZ1aSUyRmNvbXBvbmVudHMlMkZteS1nbGltbWVyLWFwcCUyRmNvbXBvbmVudC50cyUyMiUyQyUyMnNvdXJjZVRleHQlMjIlM0ElMjJpbXBvcnQlMjBDb21wb25lbnQlMkMlMjAlN0IlMjB0cmFja2VkJTIwJTdEJTIwZnJvbSUyMCclNDBnbGltbWVyJTJGY29tcG9uZW50JyUzQiU1Q25leHBvcnQlMjBkZWZhdWx0JTIwY2xhc3MlMjBleHRlbmRzJTIwQ29tcG9uZW50JTIwJTdCJTVDbiUyMCUyMCU0MHRyYWNrZWQlMjB2YWx1ZXMlM0IlNUNuJTVDbiUyMCUyMGNvbnN0cnVjdG9yKG9wdGlvbnMpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMHN1cGVyKG9wdGlvbnMpJTNCJTVDbiU1Q24lMjAlMjAlMjAlMjB0aGlzLnZhbHVlcyUyMCUzRCUyMCU1QjElMkMlMjAyJTJDJTIwMyUyQyUyMDQlMkMlMjA1JTJDJTIwNiU1RCUzQiU1Q24lMjAlMjAlN0QlNUNuJTVDbiUyMCUyMGluY3JlbWVudEluZGV4KGluZGV4KSUyMCU3QiU1Q24lMjAlMjAlMjAlMjB0aGlzLnZhbHVlcyU1QmluZGV4JTVEJTIwJTJCJTNEJTIwMSUzQiU1Q24lMjAlMjAlMjAlMjB0aGlzLnZhbHVlcyUyMCUzRCUyMHRoaXMudmFsdWVzJTNCJTVDbiUyMCUyMCU3RCU1Q24lN0QlMjIlN0QlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyaW5uZXItY29tcG9uZW50JTIyJTJDJTIydGVtcGxhdGUlMjIlM0ElN0IlMjJmaWxlTmFtZSUyMiUzQSUyMnNyYyUyRnVpJTJGY29tcG9uZW50cyUyRmlubmVyLWNvbXBvbmVudCUyRnRlbXBsYXRlLmhicyUyMiUyQyUyMnNvdXJjZVRleHQlMjIlM0ElMjIlM0NkaXYlM0UlNUNuJTIwJTIwSXMlMjAlN0IlN0IlNDB2YWx1ZSU3RCU3RCUyMCU1Q24lMjAlMjAlM0NidXR0b24lMjBvbmNsaWNrJTNEJTdCJTdCYWN0aW9uJTIwJTQwb25DbGljayU3RCU3RCUzRSUyQiUzQyUyRmJ1dHRvbiUzRSUyMCU1Q24lMjAlMjBldmVuJTNGJTIwJTNEJTNEJTNFJTIwJTdCJTdCaXNFdmVuJTdEJTdEJTIwJTVDbiUzQyUyRmRpdiUzRSUyMiU3RCUyQyUyMmNvbXBvbmVudCUyMiUzQSU3QiUyMmZpbGVOYW1lJTIyJTNBJTIyc3JjJTJGdWklMkZjb21wb25lbnRzJTJGaW5uZXItY29tcG9uZW50JTJGY29tcG9uZW50LnRzJTIyJTJDJTIyc291cmNlVGV4dCUyMiUzQSUyMmltcG9ydCUyMENvbXBvbmVudCUyQyUyMCU3QiUyMHRyYWNrZWQlMjAlN0QlMjBmcm9tJTIwJTVDJTIyJTQwZ2xpbW1lciUyRmNvbXBvbmVudCU1QyUyMiUzQiU1Q24lNUNuZXhwb3J0JTIwZGVmYXVsdCUyMGNsYXNzJTIwZXh0ZW5kcyUyMENvbXBvbmVudCUyMCU3QiU1Q24lMjAlMjAlNDB0cmFja2VkKCdhcmdzJyklMjBnZXQlMjBpc0V2ZW4oKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjBjb25zb2xlLmxvZyh0aGlzLmFyZ3MudmFsdWUpJTNCJTVDbiUyMCUyMCUyMCUyMHJldHVybiUyMHRoaXMuYXJncy52YWx1ZSUyMCUyNSUyMDIlMjAlM0QlM0QlM0QlMjAwJTNCJTVDbiUyMCUyMCU3RCU1Q24lN0QlM0IlMjIlN0QlN0QlNUQ=)
 * [Fixed in 0.8](http://playground.glimmerjs.com/?app=JTdCJTIyY29tcG9uZW50cyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJHbGltbWVyQXBwJTIyJTJDJTIydGVtcGxhdGUlMjIlM0ElN0IlMjJmaWxlTmFtZSUyMiUzQSUyMnNyYyUyRnVpJTJGY29tcG9uZW50cyUyRkdsaW1tZXJBcHAlMkZ0ZW1wbGF0ZS5oYnMlMjIlMkMlMjJzb3VyY2VUZXh0JTIyJTNBJTIyJTNDZGl2JTNFJTVDbiUyMCUyMCUzQ2gxJTNFRml4ZWQlMjBpbiUyMCU0MGdsaW1tZXIlMkZhcHBsaWNhdGlvbiUyMDAuOC4wJTNDJTJGaDElM0UlNUNuJTIwJTIwJTdCJTdCJTIzZWFjaCUyMHZhbHVlcyUyMGtleSUzRCU1QyUyMiU0MGluZGV4JTVDJTIyJTIwYXMlMjAlN0N2YWx1ZSUyMGluZGV4JTdDJTdEJTdEJTVDbiUyMCUyMCUyMCUyMCUzQ0lubmVyQ29tcG9uZW50JTVDbiUyMCUyMCUyMCUyMCUyMCUyMCU0MHZhbHVlJTNEJTdCJTdCdmFsdWUlN0QlN0QlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTQwb25DbGljayUzRCU3QiU3QmFjdGlvbiUyMGluY3JlbWVudEluZGV4JTIwaW5kZXglN0QlN0QlNUNuJTIwJTIwJTIwJTIwJTJGJTNFJTVDbiUyMCUyMCU3QiU3QiUyRmVhY2glN0QlN0QlNUNuJTNDJTJGZGl2JTNFJTIyJTdEJTJDJTIyY29tcG9uZW50JTIyJTNBJTdCJTIyZmlsZU5hbWUlMjIlM0ElMjJzcmMlMkZ1aSUyRmNvbXBvbmVudHMlMkZHbGltbWVyQXBwJTJGY29tcG9uZW50LnRzJTIyJTJDJTIyc291cmNlVGV4dCUyMiUzQSUyMmltcG9ydCUyMENvbXBvbmVudCUyQyUyMCU3QiUyMHRyYWNrZWQlMjAlN0QlMjBmcm9tJTIwJyU0MGdsaW1tZXIlMkZjb21wb25lbnQnJTNCJTVDbmV4cG9ydCUyMGRlZmF1bHQlMjBjbGFzcyUyMGV4dGVuZHMlMjBDb21wb25lbnQlMjAlN0IlNUNuJTIwJTIwJTQwdHJhY2tlZCUyMHZhbHVlcyUzQiU1Q24lNUNuJTIwJTIwY29uc3RydWN0b3Iob3B0aW9ucyklMjAlN0IlNUNuJTIwJTIwJTIwJTIwc3VwZXIob3B0aW9ucyklM0IlNUNuJTVDbiUyMCUyMCUyMCUyMHRoaXMudmFsdWVzJTIwJTNEJTIwJTVCMSUyQyUyMDIlMkMlMjAzJTJDJTIwNCUyQyUyMDUlMkMlMjA2JTVEJTNCJTVDbiUyMCUyMCU3RCU1Q24lNUNuJTIwJTIwaW5jcmVtZW50SW5kZXgoaW5kZXgpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMHRoaXMudmFsdWVzJTVCaW5kZXglNUQlMjAlMkIlM0QlMjAxJTNCJTVDbiUyMCUyMCUyMCUyMHRoaXMudmFsdWVzJTIwJTNEJTIwdGhpcy52YWx1ZXMlM0IlNUNuJTIwJTIwJTdEJTVDbiU3RCUyMiU3RCU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJJbm5lckNvbXBvbmVudCUyMiUyQyUyMnRlbXBsYXRlJTIyJTNBJTdCJTIyZmlsZU5hbWUlMjIlM0ElMjJzcmMlMkZ1aSUyRmNvbXBvbmVudHMlMkZJbm5lckNvbXBvbmVudCUyRnRlbXBsYXRlLmhicyUyMiUyQyUyMnNvdXJjZVRleHQlMjIlM0ElMjIlM0NkaXYlM0UlNUNuJTIwJTIwSXMlMjAlN0IlN0IlNDB2YWx1ZSU3RCU3RCUyMCU1Q24lMjAlMjAlM0NidXR0b24lMjBvbmNsaWNrJTNEJTdCJTdCYWN0aW9uJTIwJTQwb25DbGljayU3RCU3RCUzRSUyQiUzQyUyRmJ1dHRvbiUzRSUyMCU1Q24lMjAlMjBldmVuJTNGJTIwJTNEJTNEJTNFJTIwJTdCJTdCaXNFdmVuJTdEJTdEJTIwJTVDbiUzQyUyRmRpdiUzRSUyMiU3RCUyQyUyMmNvbXBvbmVudCUyMiUzQSU3QiUyMmZpbGVOYW1lJTIyJTNBJTIyc3JjJTJGdWklMkZjb21wb25lbnRzJTJGSW5uZXJDb21wb25lbnQlMkZjb21wb25lbnQudHMlMjIlMkMlMjJzb3VyY2VUZXh0JTIyJTNBJTIyaW1wb3J0JTIwQ29tcG9uZW50JTJDJTIwJTdCJTIwdHJhY2tlZCUyMCU3RCUyMGZyb20lMjAlNUMlMjIlNDBnbGltbWVyJTJGY29tcG9uZW50JTVDJTIyJTNCJTVDbiU1Q25leHBvcnQlMjBkZWZhdWx0JTIwY2xhc3MlMjBleHRlbmRzJTIwQ29tcG9uZW50JTIwJTdCJTVDbiUyMCUyMCU0MHRyYWNrZWQoJ2FyZ3MnKSUyMGdldCUyMGlzRXZlbigpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMGNvbnNvbGUubG9nKHRoaXMuYXJncy52YWx1ZSklM0IlNUNuJTIwJTIwJTIwJTIwcmV0dXJuJTIwdGhpcy5hcmdzLnZhbHVlJTIwJTI1JTIwMiUyMCUzRCUzRCUzRCUyMDAlM0IlNUNuJTIwJTIwJTdEJTVDbiU3RCUzQiUyMiU3RCU3RCU1RCUyQyUyMmhlbHBlcnMlMjIlM0ElNUIlNUQlN0Q=)
